src/main/java/com/store/book/service/shoppingcart/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,95 @@
+package com.store.book.service.shoppingcart.impl;
+
+import com.store.book.dto.cartitem.CartItemDto;
+import com.store.book.dto.cartitem.CreateCartItemRequestDto;
+import com.store.book.dto.cartitem.UpdateCartItemRequestDto;
+import com.store.book.dto.shoppingcart.ShoppingCartDto;
+import com.store.book.exception.EntityNotFoundException;
+import com.store.book.mapper.CartItemMapper;
+import com.store.book.mapper.ShoppingCartMapper;
+import com.store.book.model.Book;
+import com.store.book.model.CartItem;
+import com.store.book.model.ShoppingCart;
+import com.store.book.model.User;
+import com.store.book.repository.book.BookRepository;
+import com.store.book.repository.cartitem.CartItemRepository;
+import com.store.book.repository.shoppingcart.ShoppingCartRepository;
+import com.store.book.service.shoppingcart.ShoppingCartService;
+import com.store.book.service.user.UserService;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final BookRepository bookRepository;
+    private final UserService userService;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final CartItemMapper cartItemMapper;
+    private final ShoppingCartMapper shoppingCartMapper;
+
+    @Transactional
+    @Override
+    public ShoppingCartDto findShoppingCartByUser() {
+        Long userId = userService.getCurrentUser().get().getId();
+        ShoppingCart shoppingCart = getShoppingCartByUserId(userId);
+        ShoppingCartDto dto = shoppingCartMapper.toDto(shoppingCart);
+        return shoppingCartMapper.toDto(shoppingCart);
+    }
+
+    @Transactional
+    @Override
+    public CartItemDto addCartItem(CreateCartItemRequestDto requestDto) {
+        User user = userService.getCurrentUser().get();
```

Comment:
using Optional#get without previous Optional#isPresent check is bad practice and may cause NPE. 

think about using Optional#orElse, Optional#orElseThrow